src/main/java/zmq/Ctx.java

```diff
@@ -344,6 +346,7 @@ public void destroySocket(SocketBase socket)
         try {
             int tid = socket.getTid();
             emptySlots.add(tid);
+            slots[tid].close();
```

Comment:
Seems like we don't want to revert this fix. I would advocate for removing this commit.